src/main/java/teammates/ui/controller/AdminSessionsPageAction.java

```diff
@@ -146,17 +142,18 @@ private ActionResult createShowPageResultIfParametersInvalid() {
             statusToAdmin = "Admin Sessions Page Load<br>"
                           + "<span class=\"bold\"> Error: Missing Parameters</span>";
 
-            prepareDefaultPageData(calStart, calEnd);
+            prepareDefaultPageData(TimeHelper.convertLocalDateTimeToInstant(start, zone),
+                    TimeHelper.convertLocalDateTimeToInstant(end, zone));
             data.init(this.map, this.sessionToInstructorIdMap, this.totalOngoingSessions,
                       this.totalOpenStatusSessions, this.totalClosedStatusSessions, this.totalWaitToOpenStatusSessions,
-                      this.totalInstitutes, this.rangeStart, this.rangeEnd, this.zone, this.isShowAll);
+                      this.totalInstitutes, this.rangeStart, this.rangeEnd, this.timeZone, this.isShowAll);
             return createShowPageResult(Const.ViewURIs.ADMIN_SESSIONS, data);
```

Comment:
Most of these lines are the same as in some if-else block above. I think we should refactor these if-else or bring this out to a method to eliminate this code duplication.

Reply:
Yeah... I refactored in place so that I could worry about breaking less things, so all the original logic is still here in its glory. Let me see what I can do.